homeassistant/components/sensor/zwave.py

```diff
@@ -18,7 +18,7 @@
 def get_device(node, values, **kwargs):
     """Create zwave entity device."""
     # Generic Device mappings
-    if values.primary.command_class == zwave.const.COMMAND_CLASS_BATTERY:
+    if node.has_command_class(zwave.const.COMMAND_CLASS_BATTERY):
```

Comment:
The light discovery is a little different, since we want to change the class that is created if it has color support at all.